lib/segment/src/index/field_index/numeric_index.rs

```diff
@@ -325,8 +325,13 @@ impl<T: Encodable + Numericable> PayloadFieldIndex for NumericIndex<T> {
     fn filter(
         &self,
         condition: &FieldCondition,
-    ) -> Option<Box<dyn Iterator<Item = PointOffsetType> + '_>> {
-        let cond_range = condition.range.as_ref()?;
+    ) -> OperationResult<Box<dyn Iterator<Item = PointOffsetType> + '_>> {
+        let cond_range = condition
+            .range
+            .as_ref()
+            .ok_or(OperationError::service_error(
+                "failed to get condition range",
+            ))?;
```

Comment:
We probably want `ok_or_else` here to prevent constructing the error if not needed.